SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,257 @@
+            _vis3.Size = new Vector2((float)Rectangle3.ActualWidth, (float)Rectangle3.ActualHeight);
+
+            _vis4.Brush = _brush1;
+            _vis4.Scale = new Vector3(0, 1, 0);
+            _vis4.Size = new Vector2((float)Rectangle4.ActualWidth, (float)Rectangle4.ActualHeight);
+
+            // Parent visuals to XAML rectangles
+            ElementCompositionPreview.SetElementChildVisual(Rectangle1, _vis1);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle2, _vis2);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle3, _vis3);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle4, _vis4);
+
+
+            //create the scale & offset animation
+            Vector3KeyFrameAnimation offsetAnim_r1 = _compositor.CreateVector3KeyFrameAnimation();
+            offsetAnim_r1.InsertKeyFrame(1, new Vector3((float)Rectangle1.ActualWidth, (float)Rectangle1.ActualHeight, 0));
+            offsetAnim_r1.Duration = TimeSpan.FromSeconds(1);
+
+            _scaleAnim = _compositor.CreateVector3KeyFrameAnimation();
+            _scaleAnim.InsertKeyFrame(0, new Vector3(0, 1, 0));
+            _scaleAnim.InsertKeyFrame(.5f, new Vector3(1, 1, 0));
+            _scaleAnim.InsertKeyFrame(1, new Vector3(0, 1, 0));
+            _scaleAnim.Duration = TimeSpan.FromSeconds(2);
+
+
+            // animation of color stops
+            _colorAnimBrush1Stop1 = _compositor.CreateColorKeyFrameAnimation();
+            _colorAnimBrush1Stop1.InsertKeyFrame(.5f, Colors.Honeydew);
+            _colorAnimBrush1Stop1.Duration = TimeSpan.FromSeconds(4);
+
+            _colorAnimBrush1Stop2 = _compositor.CreateColorKeyFrameAnimation();
+            _colorAnimBrush1Stop2.InsertKeyFrame(.5f, Colors.DeepPink);
+            _colorAnimBrush1Stop2.Duration = TimeSpan.FromSeconds(4);
+
+            _offsetAnim = _compositor.CreateScalarKeyFrameAnimation();
+            _offsetAnim.Duration = TimeSpan.FromSeconds(1);
+
+            // when the buttons of text are pressed, the brush will change colors. Below is the set up for animation
+            _changeb1stop1 = _compositor.CreateColorKeyFrameAnimation();
+            _changeb1stop1.InsertKeyFrame(.5f, Colors.LightSkyBlue);
+            _changeb1stop1.Duration = TimeSpan.FromSeconds(2);
+
+            _changeb1stop2 = _compositor.CreateColorKeyFrameAnimation();
+            _changeb1stop2.InsertKeyFrame(.5f, Colors.Teal);
+            _changeb1stop2.Duration = TimeSpan.FromSeconds(2);
+        }
+
+        /*
+         * Run animation on target visual brush
+         */
+
+        private void Pointer_Pressed(object sender, RoutedEventArgs e)
+        {
+            pointerPressedChangeColors(b1GradientStop1, b1GradientStop2);
+        }
+
+        private void pointerPressedChangeColors(CompositionColorGradientStop target, CompositionColorGradientStop target1)
+        {
+
+            if (target.Color == Colors.DeepPink || target.Color == Colors.Honeydew)
```

Comment:
Here and elsewhere through the sample, you're using "target.Color" to choose which action to take.  One problem with this is that it makes the sample more challenging to understand.   What is the relationship between target.Color being DeepPink and the fact that you should use _changeb1stop1 rather than _colorAnimBrush1Stop1?  Perhaps you should have a separate member variable which indicates what state you're in, and then you can give it a clearer name that makes the logic more obvious (ex _animatingFromLightToDark)

Reply:
I think I resolved this comment in my most recent update, let me know if it is still confusing to read